plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ApplyMojo.java

```diff
@@ -439,8 +417,9 @@ protected void applyEntities(Controller controller, KubernetesClient kubernetes,
 
         File file = null;
         try {
-            file = getKubeCtlExecutable(controller);
-        } catch (MojoExecutionException e) {
+            Fabric8ServiceHub hub = getFabric8ServiceHub(controller);
+            file = hub.getClientToolsService().getKubeCtlExecutable();
+        } catch (Exception e) {
             log.warn("%s", e.getMessage());
         }
         if (file != null) {
```

Comment:
[2] As the whole thing here is only used to print out the proper command (`oc` or `kubect`) for the info message (which btw should be _info_, not _warn_), I would simply check for the fabric8.mode. I think this is good enough and you could remove all the dance in this method to get the file.